### ISSUE TITLE
Macro Validation

### ISSUE THREAD

**Issue by h3rald** [h3rald/glyph#94]
It should be possible to specify whether a macro is allowed within another macro or not or similar.

Example:

```
    macro :test do
      validate { @node.parent[:macro]== :a }
      # macro code
    end
```

**Comment by h3rald** [comment]
DOC: New MacroError -- Macro '#@name' is not allowed here.

**Comment by h3rald** [comment]
Working on this.

**Comment by h3rald** [comment]
Implemented custom macro validation.
- Closed by eb6669da44e1e5ab552be7fd19a99f5cfe7b6b0e.
